lighthouse-core/gather/driver/network-monitor.js

```diff
@@ -0,0 +1,247 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const log = require('lighthouse-logger');
+const {EventEmitter} = require('events');
+const NetworkRecorder = require('../../lib/network-recorder.js');
+const {NON_NETWORK_PROTOCOLS} = require('../../lib/url-shim.js');
+
+/** @typedef {import('../../lib/network-recorder.js').NetworkRecorderEvent} NetworkRecorderEvent */
+/** @typedef {import('../../lib/network-request.js')} NetworkRequest */
+/** @typedef {'network-2-idle'|'network-critical-idle'|'networkidle'|'networkbusy'|'network-critical-busy'|'network-2-busy'} NetworkMonitorEvent_ */
+/** @typedef {NetworkRecorderEvent|NetworkMonitorEvent_} NetworkMonitorEvent */
+
+const IGNORED_NETWORK_SCHEMES = [...NON_NETWORK_PROTOCOLS, 'ws'];
+
+/**
+ * @fileoverview This class wires up the procotol to a network recorder and provides overall
```

Comment:
File overview comments should be at the top (right after use strict)

Reply:
If so, we are rather inconsistent in this regard :)

Reply:
moved

Reply:
ya :( 

I'm gonna go with the nuclear option https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-rules-require-file-overview

Reply:
If we care enough we should have put a linter rule on it :)

Honestly we often use `@fileoverview` as more of a class comment and we have no tools that consume them, so the exact placement at the top doesn't seem that important. I know I've haven't ever been bothered by people's locations, so "should" seems a little strong.

Rather than having this be a human review thing, we could open an issue to consider turning on the jsdoc eslint plugin (notably they don't include `require-file-overview` in their recommended rules :P), but more important things like `require-returns` might be controversial :)

Reply:
+1 to if we care enough, in the future this should be a linter thing :)

I was in fact treating this one as a class comment like Brendan described which is why I put it immediately above the class.